File: BookNestFrontend/booknest_desktop/lib/models/notification_type.dart
class NotificationType {
  final int id;
  final String name;

  NotificationType({required this.id, required this.name});

  factory NotificationType.fromJson(Map<String, dynamic> json) {
    return NotificationType(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}

File: BookNestBackend/BookNest.Services/Database/Entities/NotificationType.cs
using System.ComponentModel.DataAnnotations;

namespace BookNest.Services.Database.Entities
{
    public class NotificationType
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
    }
}
